### DISCUSSION TITLE
Smplayer & vlc is running inside firejail when launched from KDE menu but running outside of firejail when I click on a video file

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#5074] [Q&A] [answered]
Hi,
Two days back I moved from Arch to Debian. I have used `sudo firecfg`. When I go to the KDE menu and click on smplayer or vlc and check using the command `firejail --list` I see both smplayer and vlc.

The problem is instead of going to the menu if I right click on any audio/video file and select open with smplayer or vlc and then again check using the command` firejail --list` I don't see smplayer/vlc.

Its quite troublesome to open smplayer/vlc from the menu every time I want to watch a video or listen to a song. 

How do I fix this ?

**Comment by rusty-snake** [comment, marked as answer]
Does `firejail --tree` show them? Or better do you firejail dolphin? Because you can not stack firejail sandboxes.

**Reply by firejailaddssecuirty** [reply]
These are after clicking a video file and playing it with smplayer
```
$ firejail --list
19764:home::firejail .local/opt/firefox/firefox -P 
20706:home::firejail .local/opt/firefox/firefox -P 
21213:home:dolphin:/usr/bin/firejail /usr/bin/dolphin 
```
```
$ firejail --tree
19764:home::firejail .local/opt/firefox/firefox -P 
  19765:home::/usr/bin/xdg-dbus-proxy --fd=8 --args=9 
  19771:home::firejail .local/opt/firefox/firefox -P 
    19780:home::.local/opt/firefox/firefox -P 
      19897:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -parentBuildID 20220322144853 -prefsLen 1 -prefMapSize 247183 -appDir /home/home/.local/opt/firefox/browser 9 true socket 
      19921:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 1 -isForBrowser -prefsLen 139 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      19957:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 2 -isForBrowser -prefsLen 5131 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20031:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 3 -isForBrowser -prefsLen 5830 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20053:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 5 -isForBrowser -prefsLen 5830 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20582:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -parentBuildID 20220322144853 -prefsLen 8498 -prefMapSize 247183 -appDir /home/home/.local/opt/firefox/browser 9 true rdd 
      21102:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 9 -isForBrowser -prefsLen 8637 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      21190:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 10 -isForBrowser -prefsLen 8637 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      21416:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 11 -isForBrowser -prefsLen 8637 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      21599:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 12 -isForBrowser -prefsLen 8853 -prefMapSize 247183 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
20706:home::firejail .local/opt/firefox/firefox -P 
  20711:home::/usr/bin/xdg-dbus-proxy --fd=8 --args=9 
  20713:home::firejail .local/opt/firefox/firefox -P 
    20726:home::.local/opt/firefox/firefox -P 
      20904:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 2 -isForBrowser -prefsLen 5040 -prefMapSize 255017 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20966:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 3 -isForBrowser -prefsLen 5739 -prefMapSize 255017 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20978:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 4 -isForBrowser -prefsLen 5739 -prefMapSize 255017 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      20992:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 5 -isForBrowser -prefsLen 5739 -prefMapSize 255017 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
      21065:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -parentBuildID 20220322144853 -prefsLen 6168 -prefMapSize 255017 -appDir /home/home/.local/opt/firefox/browser 9 true rdd 
      21082:home::/home/home/.local/opt/firefox/firefox-bin -contentproc -childID 6 -isForBrowser -prefsLen 6168 -prefMapSize 255017 -jsInitLen 279424 -parentBuildID 20220322144853 -appDir /home/home/.local/opt/firefox/browser 9 true tab 
21213:home:dolphin:/usr/bin/firejail /usr/bin/dolphin 
  21215:home::/usr/bin/firejail /usr/bin/dolphin 
    21218:home::/usr/bin/dolphin 
      21326:home::/usr/bin/smplayer /media/home/33216df5-95c3-456c-9230-6583ab329348/Full Video -Ghoomar_Padmaavat_Deepika Padukone Shahid Kapoor Ranveer Singh_Shreya Ghoshal SwaroopKhan [CU1tFtk_NFY].webm 
        21339:home::/usr/bin/mpv --no-config --no-quiet --terminal --no-msg-color --input-ipc-server=/tmp/smplayer-mpv-56 --msg-level=ffmpeg/demuxer=error --no-fs --hwdec=vaapi --sub-auto=fuzzy --no-input-default-bindings --input-vo-keyboard=no --no-input-cursor --cursor-autohide=no --no-keepaspect --wid=73400345 --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --vid=1 --sub-pos=100 --volume=110 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/home/home/Pictures/smplayer_screenshots --audio-pitch-correction=yes --volume-max=110 --term-playing-msg=MPV_VERSION=${=mpv-version:}
INFO_VIDEO_WIDTH=${=width}
INFO_VIDEO_HEIGHT=${=height}
INFO_VIDEO_ASPECT=${=video-params/aspect}
INFO_VIDEO_FPS=${=container-fps:${=fps}}
INFO_VIDEO_FORMAT=${=video-format}
INFO_VIDEO_CODEC=${=video-codec}
INFO_AUDIO_FORMAT=${=audio-codec-name}
INFO_AUDIO_CODEC=${=audio-codec}
INFO_AUDIO_RATE=${=audio-params/samplerate}
INFO_AUDIO_NCH=${=audio-params/channel-count}
INFO_LENGTH=${=duration:${=length}}
INFO_DEMUXER=${=current-demuxer:${=demuxer}}
INFO_SEEKABLE=${=seekable}
INFO_TITLES=${=disc-titles}
INFO_CHAPTERS=${=chapters}
INFO_TRACKS_COUNT=${=track-list/count}
METADATA_TITLE=${metadata/by-key/title:}
METADATA_ARTIST=${metadata/by-key/artist:}
METADATA_ALBUM=${metadata/by-key/album:}
METADATA_GENRE=${metadata/by-key/genre:}
METADATA_DATE=${metadata/by-key/date:}
METADATA_TRACK=${metadata/by-key/track:}
METADATA_COPYRIGHT=${metadata/by-key/copyright:}
INFO_MEDIA_TITLE=${=media-title:}
INFO_STREAM_PATH=${stream-path}
 --audio-client-name=SMPlayer --term-status-msg=STATUS: ${=time-pos} / ${=duration:${=length:0}} P: ${=pause} B: ${=paused-for-cache} I: ${=core-idle} VB: ${=video-bitrate:0} AB: ${=audio-bitrate:0} /media/home/33216df5-95c3-456c-9230-6583ab329348/Full Video -Ghoomar_Padmaavat_Deepika Padukone Shahid Kapoor Ranveer Singh_Shreya Ghoshal SwaroopKhan [CU1tFtk_NFY].webm 
    21232:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/tags.so tags  local:/run/user/1000/dolphinSGkNoX.1.slave-socket 
    21234:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/trash.so trash  local:/run/user/1000/dolphinAguOaS.2.slave-socket 
    21236:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/file.so file  local:/run/user/1000/dolphinBTtRHW.3.slave-socket 
    21239:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/file.so file  local:/run/user/1000/dolphinBiDbdT.4.slave-socket 
    21241:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/file.so file  local:/run/user/1000/dolphinvaesWz.5.slave-socket 
    21243:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/file.so file  local:/run/user/1000/dolphinYfGAfd.6.slave-socket 
    21245:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/file.so file  local:/run/user/1000/dolphinKIJBOB.7.slave-socket 
    21247:home::/usr/lib/x86_64-linux-gnu/libexec/kf5/kioslave5 /usr/lib/x86_64-linux-gnu/qt5/plugins/kf5/kio/thumbnail.so thumbnail  local:/run/user/1000/dolphinGbEAYS.8.slave-socket 
```
I am pretty sure when I used to click on a video file under Arch and then check with `firejail --list` I used to see smplayer

**Reply by firejailaddssecuirty** [reply]
> Because you can not stack firejail sandboxes.

What does that mean ?

I want both vlc and smplayer to run inside a sandbox. The reason for that is I have created profiles in my `~/.config/firejail` for them with the `net none` option.

Why would anyone want to sandbox their file manager ? What kind of attacks does it prevent ? I mean file managers like dolphin, thunar, pcmanfm are not internet facing as far as I know.

**Reply by rusty-snake** [reply]
> ```
> 21213:home:dolphin:/usr/bin/firejail /usr/bin/dolphin 
>  21215:home::/usr/bin/firejail /usr/bin/dolphin 
>    21218:home::/usr/bin/dolphin 
>      21326:home::/usr/bin/smplayer /media/home/33216df5-95c3-456c-9230-6583ab329348/Full Video -Ghoomar_Padmaavat_Deepika Padukone Shahid Kapoor Ranveer Singh_Shreya Ghoshal SwaroopKhan [CU1tFtk_NFY].webm 
>        21339:home::/usr/bin/mpv --no-config --no-quiet --terminal --no-msg-color --input-ipc-server=/tmp/smplayer-mpv-56 --msg-level=ffmpeg/demuxer=error --no-fs --hwdec=vaapi --sub-auto=fuzzy --no-input-default-bindings --input-vo-keyboard=no --no-input-cursor --cursor-autohide=no --no-keepaspect --wid=73400345 --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --vid=1 --sub-pos=100 --volume=110 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/home/home/Pictures/smplayer_screenshots --audio-pitch-correction=yes --volume-max=110 --term-playing-msg=MPV_VERSION=${=mpv-version:}
> ```

smplayer is running in the dolphin sandbox.

**Reply by rusty-snake** [reply]
> What does that mean ?

If you run `firejail some-program` inside a firejail sandbox, `some-program` will run in the sandbox of first program.

**Reply by rusty-snake** [reply]
> I want both vlc and smplayer to run inside a sandbox.

They do, but it is the dolphin sandbox.

> The reason for that is I have created profiles in my ~/.config/firejail for them with the net none option.

The you must remove dolphin from firecfg handling.

- `sudo rm /usr/local/bin/dolphin`
- `sudo sed "s/dolphin/#dolphin/" /path/to/firecfg.config`

**Reply by firejailaddssecuirty** [reply]
> The you must remove dolphin from firecfg handling.
> 
>     * `sudo rm /usr/local/bin/dolphin`
> 
>     * `sudo sed "s/dolphin/#dolphin/" /path/to/firecfg.config`

How to I know which the exact path of `/path/to/firecfg.config `in my case ?

**Reply by rusty-snake** [reply]
> I mean file managers like dolphin, thunar, pcmanfm are not internet facing as far as I know.

dolphin supports smb, ssh and ftp IIRC.

> Why would anyone want to sandbox their file manager ? What kind of attacks does it prevent ?

The most media libraries are complex, not of high quality written in memory unsafe languages ... short they are a security risk with untrusted input.

All file-managers (except of non-flatpaked nautilus in recent versions) run their thumbnailers unsanboxed. That means if thumbnails are enabled and you open a folder with a malicious picture/video you can be compromised (advanced attack if you keep your software up-to-date).

However since a file-manager needs to start all programs, view all files, talk to the DE (for desktop files, ...) the sandbox is very weak and you are probably better in sandboxing the actual programs you use to view the files and disable thumbnails in the file-manager.

**Reply by rusty-snake** [reply]
> How to I know which the exact path of /path/to/firecfg.config in my case ?

- firejail>=0.9.68: `SYSCONFIGDIR/firejail/firecfg.config` (`/etc/firejail/firecfg.config`).
- firejail<0.9.68: `LIBDIR/firejail/firecfg.config` (`/usr/lib(64)/(x68_64_...)/firejail/firecfg.config`) use `locate`.

**Reply by firejailaddssecuirty** [reply]
In that case I want to continue with dolphin sandboxed. I will launch smplayer/vlc from the KDE menu no matter how annoying it is.

**Reply by rusty-snake** [reply]
FYI (next update possibly) https://github.com/netblue30/firejail/commit/12d1de484556f397997f61a24ab7b1f2ef0ed9a6

**Reply by firejailaddssecuirty** [reply]
> FYI (next update possibly) [12d1de4](https://github.com/netblue30/firejail/commit/12d1de484556f397997f61a24ab7b1f2ef0ed9a6)

Okay but question is will users of fixed release distros like Debian get that update from the official repos ?

**Reply by rusty-snake** [reply]
With the next release (bookwork).

**Reply by firejailaddssecuirty** [reply]
I moved from Arch to Debian only coz the Arch repos are small in comparison to Debian. I was not comfortable installing from AUR. Now Debian has another issue. Just for 1 package (firejail) I will have to move to the next release.

**Reply by rusty-snake** [reply]
>  Just for 1 package (firejail) I will have to move to the next release.

https://github.com/netblue30/firejail#debian:

> Debian stable (bullseye): We recommend to use the [backports](https://packages.debian.org/bullseye-backports/firejail) package.

**Reply by rusty-snake** [reply]
> Arch repos are small in comparison to Debian.

Small but maintained. Debian has a lot packages that are unmaintaind (for years), don't get updates, are horribly outdated, ...

**Reply by firejailaddssecuirty** [reply]
In your experience which distro has a good amount of packages and are maintained properly ? Which distro do you use ?

**Reply by rusty-snake** [reply]
- Everything is better the Ubuntu (because of universe)
- Backporting needs more work than packaging, rolling-release and short-term-release distros have better maintained repos.
- Big distros have better maintained repos than small (community) distros.
- Debian has probably the most packages in their repos
- Arch has the most packages if you count repos+AUR

I use Fedora Linux.

**Reply by firejailaddssecuirty** [reply]
Haven't used Fedora for sometime now. Downloading now ....

**Comment by kmk3** [comment]
@firejailaddssecuirty [on Mar 26](https://github.com/netblue30/firejail/discussions/5074#discussion-3964962):

> Hi, Two days back I moved from Arch to Debian. I have used `sudo firecfg`.
> When I go to the KDE menu and click on smplayer or vlc and check using the
> command `firejail --list` I see both smplayer and vlc.
> 
> The problem is instead of going to the menu if I right click on any
> audio/video file and select open with smplayer or vlc and then again check
> using the command` firejail --list` I don't see smplayer/vlc.

If you're not indeed firejailing the file manager as @rusty-snake said, what is
the output of this:

```sh
grep ^Exec ~/.local/share/applications/smplayer.desktop
grep ^Exec ~/.local/share/applications/vlc.desktop
```

**Reply by firejailaddssecuirty** [reply]
```
$ grep ^Exec ~/.local/share/applications/smplayer.desktop
grep: /home/home/.local/share/applications/smplayer.desktop: No such file or directory
```
```
$ grep ^Exec ~/.local/share/applications/vlc.desktop
Exec=vlc --started-from-file %U
```

**Reply by ghost** [reply]
> grep: /home/home/.local/share/applications/smplayer.desktop: No such file or directory

Firecfg didn't create smplayer.desktop it seems. Create one manually to see if that changes anything.

> Exec=vlc --started-from-file %U

See [VLC not starting with "--started-from-file"](https://github.com/netblue30/firejail/issues/836). It's advised there to drop that part.

**Reply by firejailaddssecuirty** [reply]
I manually added firejail to `/usr/share/applications/smplayer.desktop `but still when I click on a video file smplayer is not running inside a sandbox.
What exactly am I supposed to do with the `/usr/share/applications/vlc.desktop `?

**Comment by ghost** [comment]
In the `firejail --tree` output there's:
```
[...]
21213:home:dolphin:/usr/bin/firejail /usr/bin/dolphin
  21215:home::/usr/bin/firejail /usr/bin/dolphin
    21218:home::/usr/bin/dolphin
[...]
```

To me that suggests you are in fact running `dolphin` firejailed...

**Reply by firejailaddssecuirty** [reply]
I am using KDE after a gap of many years. Under LXQT and XFCE when I run `sudo firecfg` pcmanfm & thunar are not added to firejail. I have no idea why dolphin is getting added.

**Reply by rusty-snake** [reply]
dolphin was always special (for no good reasons) as it [was](https://github.com/netblue30/firejail/commit/12d1de484556f397997f61a24ab7b1f2ef0ed9a6) the only file-manager in firecfg.config.

**Comment by rusty-snake** [comment]
> In that case I want to continue with dolphin sandboxed. I will launch smplayer/vlc from the KDE menu no matter how annoying it is.

You can try to edit smplayer.desktop like

```
Exec=systemd-run --user firejail /usr/bin/smplayer
```
